### PR TITLE
remove partial implementation of nested search

### PR DESCRIPTION
**Pull request by ijt**
It was confusing to have redundant code paths for search.

Most likely we'll want to refer back to the code being deleted in this change when we start working on nested search again, and also if we decide to separate the business logic from `graphqlbackend`.

Test plan: I manually ran some searches with `index:yes` and `index:no`.

CC @keegancsmith, @attfarhan, @kzh

### CODE REVIEW

File: cmd/frontend/graphqlbackend/search.go [hunk 1]
```diff
@@ -739,3 +733,40 @@ func handleRepoSearchResult(common *searchResultsCommon, repoRev search.Reposito
...
+	return searcherURLs
+}
+
+func IndexedSearch() *backend.Zoekt {
+	indexedSearchOnce.Do(func() {
+		indexedSearch = &backend.Zoekt{}
+		if zoektAddr != "" {
+			indexedSearch.Client = zoektrpc.Client(zoektAddr)
+		}
+		go func() {
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/search.go]: don't do this in a go func. conf.Watch is async, except for the very first run which is exactly what you want

Comment (ijt) [reply cmd/frontend/graphqlbackend/search.go]: This is just the code that was already there in search2.go. If you don't mind, I'll make that change in a separate PR for clarity.

Comment (keegancsmith) [reply cmd/frontend/graphqlbackend/search.go]: Oops :) Looking forward to the followup PR.

File: cmd/frontend/graphqlbackend/search.go [hunk 2]
```diff
@@ -12,9 +12,13 @@ import (
...
 
-	otlog "github.com/opentracing/opentracing-go/log"
+	"github.com/sourcegraph/sourcegraph/pkg/endpoint"
+	"github.com/sourcegraph/sourcegraph/pkg/env"
+	"github.com/sourcegraph/sourcegraph/pkg/search/backend"
+
+	zoektrpc "github.com/google/zoekt/rpc"
 	"github.com/pkg/errors"
-	"github.com/sourcegraph/sourcegraph/cmd/frontend/backend"
+	sgbackend "github.com/sourcegraph/sourcegraph/cmd/frontend/backend"
```
Comment (slimsag) [inline cmd/frontend/graphqlbackend/search.go]: naming this `sgbackend` is inconsistent with every other file in our codebase

**Pull request merged**